packages/app/src/runner/events/telemetry.ts

```diff
@@ -36,4 +38,61 @@ export const addTelemetryListeners = (Cypress) => {
       // TODO: log error when client side debug logging is available
     }
   })
+
+  const recordSpan = (spanState: 'start' | 'end', command: Cypress.CommandQueue, extendRecordSpanFn: (span?: Span) => void) => {
+    try {
+      const runnable = Cypress.state('runnable')
+      const test = Cypress.state('test')
+
+      const runnableType = runnable.type === 'hook' ? runnable.hookName : runnable.type
+
+      let span: Span | undefined
+
+      const spanName = `${runnableType}: ${command.attributes.name}(${command.attributes.args.join(',')})`
+
+      if (spanState === 'start') {
+        span = telemetry.startSpan({
+          name: spanName,
+          opts: {
+            attributes: {
+              spec: runnable.invocationDetails.relativeFile,
+              test: `test:${test.fullTitle()}`,
+              'runnable-type': runnableType,
+            },
+          },
+          isVerbose: true,
+        })
+      } else {
+        span = telemetry.getSpan(spanName)
+      }
+
+      extendRecordSpanFn(span)
+    } catch (error) {
+      // TODO: log error when client side debug logging is available
+    }
+  }
+
+  Cypress.on('command:start', (command: Cypress.CommandQueue) => {
+    recordSpan('start', command, (span) => {
+      span?.setAttribute('command-name', command.attributes.name)
+    })
+  })
+
+  Cypress.on('command:end', (command: Cypress.CommandQueue) => {
+    recordSpan('end', command, (span) => {
+      span?.setAttribute('state', command.state)
+      span?.setAttribute('numLogs', command.logs?.length || 0)
+      span?.end()
+    })
+  })
+
+  Cypress.on('command:failed', (command: Cypress.CommandQueue, error: Error) => {
```

Comment:
I think the code to emit this is missing because the event doesn't exist. I had to add it something in the [command_queue](https://github.com/cypress-io/cypress/blob/990ef91a427fc9cfc111f70e4d6d5878a2671e61/packages/driver/src/cypress/command_queue.ts#L544) and the general [event bus](https://github.com/cypress-io/cypress/blob/990ef91a427fc9cfc111f70e4d6d5878a2671e61/packages/driver/src/cypress.ts#L619) to wire it up

Reply:
Addressed in  3b533acf12712907d9b78e75f2f411f9de4f1d94